src/Kafka/ManageKafkaPermissions/components/ResourcePrefix.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { composeStories } from "@storybook/testing-react";
+import * as stories from "./ResourcePrefix.stories";
+import { userEvent } from "@storybook/testing-library";
+import { render, waitForI18n } from "../../../test-utils";
+
+const { InitialState, InvalidInput, ValidInput } = composeStories(stories);
+
+describe("Resource type", () => {
+  it("should render a text input for resource prefix", async () => {
+    const onChangeValue = jest.fn();
+    const comp = render(<InitialState onChangeValue={onChangeValue} />);
+
+    await waitForI18n(comp);
+
+    expect(
+      await comp.findByPlaceholderText("Enter prefix")
+    ).toBeInTheDocument();
+    expect(await comp.queryByText("Required")).not.toBeInTheDocument();
+    userEvent.type(comp.getByPlaceholderText("Enter prefix"), "pet");
+
+    expect(await comp.findByDisplayValue("pet")).toBeInTheDocument;
+  });
+  it("should show a text input with validation error ", async () => {
+    const onChangeValue = jest.fn();
+    const comp = render(<InvalidInput onChangeValue={onChangeValue} />);
+    await waitForI18n(comp);
+    expect(
+      await comp.findByPlaceholderText("Enter prefix")
+    ).toBeInTheDocument();
+    expect(await comp.findByText("Required")).toBeInTheDocument();
```

Comment:
That's a great question. 

If we want to show all the variations of validation for an input at the input component level, rather than a parent form level, does that include all the different validation messages that could display? E.g. "Required" and other messages about invalid characters? 

I think that could simplify navigation and make it easier to find specific examples (e.g. how does this component look if I type the wrong character?)

Reply:
@riccardo-forina  we are showing the 'Required' validation error message in the resource type select as well.
@jgiardino  Currently there is no validation if a user enters a wrong character,only if he does not give any value and clicks submit will the validation error and error message 'Required' show up
I can add validations in the component itself if the user does enter a value that is not possible and show an error. If a user assigns permission to a topic resource which starts with 'cat' , and there is no topic in the instance that starts with cat, how would we treat that? Entering an invalid value could be treated the same way as entering a value that does not match any topics 
If we do want to show an error, in that case, what should the validation message be?

Reply:
Apologies for missing this comment before. Feel free to ping me if I don't respond to comments in your PR.

There is an open jira that describes the current issues we have related to how we handle error messages: https://issues.redhat.com/browse/MGDSTRM-7502

In that jira, there's [a document](https://docs.google.com/document/d/1ITKLM-IMLi7PpZdGAGKQ8lseQHr-PB62-JgYs-FsxoI/edit) that lists all the error validation messages aside from "Required" that should display.

Most of the reasons for displaying errors are due to users entering characters that can't possibly match a topic name. E.g. they enter invalid characters, or they specify that the topic name is "." or ".." which we don't allow. 

> Currently there is no validation if a user enters a wrong character,only if he does not give any value and clicks submit will the validation error and error message 'Required' show up

If you look at what exists in production, currently the validation messages display on submit for the original Add Permission row. The behavior is different for the new shortcut options as noted in the jira. The following is a screencap of what we have in production.

![image](https://user-images.githubusercontent.com/21063328/160723313-b050ec82-00ac-492f-b972-dca8abe91f39.png)



> If a user assigns permission to a topic resource which starts with 'cat' , and there is no topic in the instance that starts with cat, how would we treat that?

We would not block that. Showing the menu supports use cases like the following:
* User wants to enter a prefix "sales-" but as they type, they realize all the topic prefixes are actually "sale-" when they see the list of existing topic names. In this case, they don't ever select a menu option, but specify a new value "sale-" and select the option 'Use "sale-"' that displays in the menu.
* User wants to specify a name of a topic that exists, so they start typing part of the name until they see the option they want.

The reason why we would not block the user from specifying "cat" if no topic exists with that name is for the following use case:
* The user wants to give an account permissions to create a topic with the name "cat". In this case, the topic "cat" doesn't exist yet, so the user would type "cat" and see the menu option 'Use "cat"' display in the menu.